### PR TITLE
Use koa v2 middleware signature + add yarn.lock

### PR DESCRIPTION
**Pull request by jrowlingson**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,14 +25,12 @@
   },
   "homepage": "https://github.com/philipheinser/ember-lightning",
   "dependencies": {
-    "co": "^4.6.0",
-    "co-redis": "^2.1.0",
-    "hiredis": "^0.4.1",
```
Comment (jrowlingson) [inline package.json:30]: ~~unused?~~

File: package.json [hunk 2]
```diff
@@ -25,14 +25,12 @@
...
-    "koa": "^2.0.0-alpha.2",
-    "redis": "^2.6.0-1"
+    "co-redis": "^2.1.1",
+    "koa": "^2.3.0",
+    "redis": "^2.7.1"
   },
   "engines": {
-    "node": "4.2.x"
+    "node": ">=7.6.0"
```
Comment (jrowlingson) [inline package.json:34]: [required](https://github.com/koajs/koa#installation) by koa for async function support.

**Comment by philipheinser** [comment]
@jrowlingson Is this ready to merge?

**Comment by jrowlingson** [comment]
@philipheinser tests are passing so this should be good to go. I also have removed the `hiredis` dep as it appeared to be unused (unless I am mistaken).

**Comment by philipheinser** [comment]
@jrowlingson I the redis client works without "hiredis" it uses an javascript client library then, but when it finds a "hiredis" package it uses the "hiredis" native implementation witch in many cases is much faster.

**Comment by jrowlingson** [comment]
@philipheinser ah, thanks for the clarification - i have reverted the removal of the dep. this should be good to merge now.

**Comment by philipheinser** [comment]
@jrowlingson Thanks!

**Pull request merged**